### ISSUE TITLE
Rule proposal: no useless expressions

### ISSUE THREAD

**Issue by dcousens** [eslint/eslint#7642]
### Warns about a potential error

I recently had something bite me in some of my own code, and I wonder if we can't avoid this happening.

``` js
var o = {}

// ...

if (o.value === undefined) {
  o.value === 3
}
```

`o.value` may not be a useless expression (could be a mutative property),  but comparing it is definitely useless without something using that return value and is almost certainly an error.

In this case, the correct code was meant to be `o.value = 3`,  but it appears repetition got the better of me that day.

**Comment by mysticatea** [comment]
Could you check [no-unused-expressions](http://eslint.org/docs/rules/no-unused-expressions)?

**Comment by dcousens** [comment]
Sorry,  thanks for the link
